### PR TITLE
Turn exercise 005 into practice

### LINKED ISSUE #2170: Turn exercise 005 into practice
Extract the OCaml code from the markdown file corresponding to this exercise. The file is: `data/exercise/005_NAME.md` (where `005` is the number from the title of this issue and `NAME` is the rest of the file name, for instance `data/exercise/001_tail.md` is the file which corresponds to exercise `001`). Use the code in this file to create a folder `practice/005` (using this same number).

Populate the `005` folder with (almost) the same files as in folders `practice/001` and `practice/002` (they are examples).

Three files are different in each folder:
1. `practice/005/ex.ml`: it contains the function's signature, which needs to be implemented by the learner; 
1. `practice/005/work/impl.ml`: it includes the template meant to be filed by the learner;
1. `practice/005/answer/impl.ml`: consists of the solution from the original markdown file.

The examples included in the original markdown file must be turned into test cases in the `practice/005/ex.ml` file.

Folder structure to create:
```
practice/005/
├── answer
│   ├── impl.ml
│   └── test
│       ├── dune
│       └── run.ml
├── dune
├── dune-project
├── ex.ml
└── work
    ├── impl.ml
    └── test
        ├── dune

### PR DESCRIPTION
**Pull request by MissJae**
This Pr is in reference to task #2170.
I am still a little confused about the task description, I would appreciate feedback on my attempt if possible @cuihtlauac 

Fix PR #2170

### CODE REVIEW

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
Thanks @MissJae, this is great.

This PR shouln't not touch folder 003, remove all the changes touching this folder from this PR.

Your work on folder 005 is good. It compiles and passes the tests. Here are the changes which are also needed:

- Add newline at the end of each file which don't
- Add an additional test case with `rev []` (call it "nil")

**Comment by cuihtlauac** [comment]
Was commit c1d45de70d7939391ad4e04a4e6e0081bca145c9 a mistake?

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
Rename folder `005.` into `005`. Add newlines at end of file. Do not use git push with force option

**Comment by MissJae** [comment]
> Was commit [c1d45de](https://github.com/ocaml/ocaml.org/commit/c1d45de70d7939391ad4e04a4e6e0081bca145c9) a mistake?

Sorry it was a mistake. I have removed it

File: practice/005./answer/impl.ml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+let rev list =
+  let rec aux acc = function
+    | [] -> acc
+    | h :: t -> aux (h :: acc) t
+  in
+  aux [] list
+
```
Comment (cuihtlauac) [inline practice/005./answer/impl.ml]:
```suggestion
```

Remove white space and add newline at end of file

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Comment by MissJae** [comment]
@cuihtlauac please can you check this? I've made the requested changes.

File: practice/005/answer/impl.ml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+let rev list =
+  let rec aux acc = function
+    | [] -> acc
+    | h :: t -> aux (h :: acc) t
+  in
+  aux [] list
+
```
Comment (cuihtlauac) [inline practice/005/answer/impl.ml]:
```suggestion
  aux [] list

```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Review by cuihtlauac — APPROVED** [review]
Thanks @MissJae

**Comment by divyankachaudhari** [comment]
@MissJae @cuihtlauac This doesn't compile.

```
~/ocaml.org/practice/005 (main ✘)✭ ᐅ dune build --root .                              
File "ex.ml", line 9, characters 42-57:
9 |     "nil" >:: (fun _ -> assert_equal None (Tested.rev []));
                                              ^^^^^^^^^^^^^^^
Error: This expression has type 'a list
       but an expression was expected of type 'b option
 ```

**Comment by MissJae** [comment]
> @MissJae @cuihtlauac This doesn't compile.
> 
> ```
> ~/ocaml.org/practice/005 (main ✘)✭ ᐅ dune build --root .                              
> File "ex.ml", line 9, characters 42-57:
> 9 |     "nil" >:: (fun _ -> assert_equal None (Tested.rev []));
>                                               ^^^^^^^^^^^^^^^
> Error: This expression has type 'a list
>        but an expression was expected of type 'b option
> ```

Hey @divyankachaudhari I think I've figured out the error. I think I need to change None to []. @cuihtlauac should I create a pull request for it?
